### ISSUE TITLE
External OAuth Provider Requests

### ISSUE THREAD

**Issue by J0** [supabase/gotrue#451]
This issue is for tracking requests/demand for client libraries. Give a comment a thumbs up if you want the connector built or drop a comment if you wish to work on any of the providers below.

**Comment by J0** [comment]
# Feature request

## Is your feature request related to a problem? Please describe.

Currently, Supabase does not support Steam as an external OAuth provider.

## Describe the solution you'd like

Support Steam as an external OAuth Provider. https://partner.steamgames.com/doc/features/auth

## Describe alternatives you've considered

N/A

## Additional context

This [article](https://owlspace.xyz/cybersec/steam-login/) describes how Steam's login method works.

**Comment by J0** [comment]
# Feature request

## Is your feature request related to a problem? Please describe.

Support Patreon oauth.

## Describe the solution you'd like

See feature request <https://github.com/netlify/gotrue/issues/312>

**Comment by J0** [comment]
Add Quickbooks as an OAuth provider

**Comment by J0** [comment]
## Add Orcid as an OAuth provider

Hi

I'd like to publish an app that other researchers can contribute to without signing up to anything, just using the orcid credentials they have for publishing to journals. Could you please add Orcid to the OAuth providers?

https://info.orcid.org/documentation/features/public-api/orcid-as-a-sign-in-option-to-your-system/

All the best and many thanks for the great work!

**Comment by J0** [comment]
## Add global.id as OAuth Provider

[Link to homepage](https://global.id)
[Link to docs](https://developer.global.id/documentation/index.html)

## Is your feature request related to a problem? Please describe.
Letting users in with a privacy-friendly OAuth Provider while accessing/verifying user data in a privacy-friendly manner if required.
For (at least currently) no cost.

## Describe the solution you'd like

Adding global.id as sign-in/up option and storing requested data in the user metadata.

## Describe alternatives you've considered

Didn't really find a good alternative to this provider.

## Additional context

I would like to implement that but I have never used go before, nor do I have a clue on how to integrate it in the existing codebase.
Also, the global.id docs are somewhat odd and I've never really dealt with implementing OAuth.
Maybe it can be done similar to using Auth0 but instead, use global.id but idk. [Article about Supabase with Auth0](https://supabase.com/docs/guides/integrations/auth0)

**Comment by J0** [comment]
# Feature request
* Sign-in with Singpass 
* https://api.singpass.gov.sg/library/identiface/developers/backend-oauth2-jws

Note: Singapore government might move to use [SGID](https://www.id.gov.sg/)

**Comment by J0** [comment]
# Feature request

## Is your feature request related to a problem? Please describe.

Would love to be able to authenticate users with Foursquare/Swarm

## Describe alternatives you've considered

Currently using [passport-foursquare](https://github.com/jaredhanson/passport-foursquare)

**Comment by J0** [comment]
# Feature request
Please add SoundCloud OAuth provider.

## Is your feature request related to a problem? Please describe.
To extend music streaming platform authentication.

## Additional context
https://developers.soundcloud.com/docs/api/guide#authentication

Note: there is an existing PR -- #269 which contains an initial implementation

**Comment by J0** [comment]
# Feature request

## Is your feature request related to a problem? Please describe.

I'd like to let my users log in with their Eth wallet (Metamask, etc)

## Describe the solution you'd like

Just like Uniswap does.

## Describe alternatives you've considered

Looks like [Redwood has an Eth login](https://github.com/oneclickdapp/ethereum-auth).

## Additional context

n/a

Note: there is an existing PR -- https://github.com/supabase/gotrue/pull/269 which contains an initial implementation

**Comment by J0** [comment]
# Feature request

## Auth0

Would it be possible to include Auth0 as a login provider. Would like transition over to Supabase however this is preventing me from doing so.

## Describe alternatives you've considered

Tutorial on importing users from Auth0.

**Comment by J0** [comment]
# Feature request

Create an OAuth provider for TikToke

TikTok recently published their OAuth flow
https://developers.tiktok.com/doc/login-kit-web

## Is your feature request related to a problem? Please describe.

For the application that I am working on, we convert users from TikTok.  Currently, we plan to authenticate them from  using Phone authentication, but TikTok support could drastically improve our conversion.

## Describe the solution you'd like

Social login with TikTok is supported similar to existing 3rd party providers.

## Describe alternatives you've considered

The only other alternative would be to host our own authentication server and use it in tandem with Supabase.  Not particularly ideal.

## Additional context

Note: there is an existing PR -- https://github.com/supabase/gotrue/pull/269 which contains an initial implementation

**Comment by J0** [comment]
# Feature request

## Is your feature request related to a problem? Please describe.

Add Netlify OAuth provider

## Additional context

https://twitter.com/jlengstorf/status/1429611357356187652

<img width="595" alt="image" src="https://user-images.githubusercontent.com/10214025/130551483-e83c85db-62e9-471f-bb37-5dea9e139959.png">

https://app.netlify.com/user/applications


Relevant Comments:
> Might need to hold off on this one until some of the security issues here are covered: https://community.redwoodjs.com/t/i-implemented-a-netlify-oauth-not-identity-auth-provider-but-im-not-sure-i-should-have-and-why/903

**Comment by J0** [comment]
## Kakao

Requested on: https://github.com/supabase/supabase/discussions/5200

Developer docs: https://developers.kakao.com/product/kakaoLogin

Corresponding PR: #366

**Comment by J0** [comment]
## Add Snapchat as External OAuth Providers #436 

Relevant PR: #449

**Comment by J0** [comment]
# Telegram

## Is your feature request related to a problem? Please describe.

For [dip.chat](https://dip.chat), we use Telegram groups. I eventually like to add login with Telegram so users can sign in to the web app and edit their profile, credit card info, etc.

## Describe the solution you'd like

As a user of supabase, I would like to be able to use [Telegram](https://telegram.org/) as an authentication provider. 

## Describe alternatives you've considered

I've requested it on NextAuth.js: https://github.com/nextauthjs/next-auth/issues/2406

I've also considered contributing it there ^ or just writing it myself in my app.

## Additional context

I am not sure where I would start, but I would considering implementing this myself with some guidance/pointers.

### Links from Telegram

- [User Authorization](https://core.telegram.org/api/auth#sign-insign-up)
- [Telegram Login Widget](https://core.telegram.org/widgets/login)
- [Telegram Login for Websites](https://telegram.org/blog/login)

Other relevant content:

> Looked at including this in #245 but as this [reply ](https://github.com/nextauthjs/next-auth/issues/2406#issuecomment-913238005) said its not standard 0Auth so the Supabase team might need to look at this one

**Comment by bariqhibat** [comment]
Hi @J0, I just looked into Steam OAuth briefly and I don't think they're giving out user's email on their API ([GetPlayerSummaries](https://partner.steamgames.com/doc/webapi/ISteamUser), and [reference](https://stackoverflow.com/questions/31571267/steam-get-users-email-address) from stackoverflow), in which I think we're in need of for our auth, what do you think?

**Comment by dangdennis** [comment]
Hi, we'd like to implement the Instagram oauth provider.

**Comment by J0** [comment]
Hey @dangdennis, 

Sorry for the delayed reply -- was a bank holiday for us over the past two days. Yes we are accepting contributions! And no worries about the frontend we will take care of it.

We'd also greatly appreciate it if you'd be able to add a short guide on how to use the provider: https://github.com/supabase/supabase/pull/5398/files after finishing the PR. This is so that developers can best figure out how to use the provider in their application. 

Feel free to let us know if you have any questions or if you run into any issues! 

Thanks so much!

**Comment by J0** [comment]
@bariqhibat 

Yup, we are accepting contributions for Yandex provider. Just saw your PR -- the team will review shortly.

Thanks much!

**Comment by Mi3liX9** [comment]
> # Telegram
> ## Is your feature request related to a problem? Please describe.
> For [dip.chat](https://dip.chat), we use Telegram groups. I eventually like to add login with Telegram so users can sign in to the web app and edit their profile, credit card info, etc.
> 
> ## Describe the solution you'd like
> As a user of supabase, I would like to be able to use [Telegram](https://telegram.org/) as an authentication provider.
> 
> ## Describe alternatives you've considered
> I've requested it on NextAuth.js: [nextauthjs/next-auth#2406](https://github.com/nextauthjs/next-auth/issues/2406)
> 
> I've also considered contributing it there ^ or just writing it myself in my app.
> 
> ## Additional context
> I am not sure where I would start, but I would considering implementing this myself with some guidance/pointers.
> 
> ### Links from Telegram
> * [User Authorization](https://core.telegram.org/api/auth#sign-insign-up)
> * [Telegram Login Widget](https://core.telegram.org/widgets/login)
> * [Telegram Login for Websites](https://telegram.org/blog/login)
> 
> Other relevant content:
> 
> > Looked at including this in #245 but as this [reply ](https://github.com/nextauthjs/next-auth/issues/2406#issuecomment-913238005) said its not standard 0Auth so the Supabase team might need to look at this one

I have no experience in Auth, but I want to contribute to building Telegram Auth API.
I have some experience in building Telegram bots, and I have the time, so where should I start and what should I do?

**Comment by J0** [comment]
Transferred: IMDb OAuth request  from #528

**Comment by kangmingtay** [comment]
# Roblox

## Is your feature request related to a problem? Please describe.

Roblox recently opened up a private alpha for their new OAuth2 system, and should become public in a few months. I can keep this issue updated with any news, so that once it becomes public it can be implemented into Supabase.

Until then, I will come up with a basic draft and see if I can get it functioning properly locally.

Roblox is what my app is based around, but I am currently using Discord OAuth through Supabase and authenticating them using other means afterwards. Adding this OAuth would make my lif etc.

## Describe the solution you'd like

As a user of supabase, I would like to be able to use Roblox as an authentication provider.

**Comment by J0** [comment]
## Reddit

## Is your feature request related to a problem? Please describe.

No.

## Describe the solution you'd like

It would be cool if Reddit OAuth could be used in Supabase.

## Describe alternatives you've considered

Making my own. 💀

## Additional context

N/A

**Comment by J0** [comment]
## Ory

I want a new Ory Hydra OAuth2 Provider Integration

## Describe the solution you'd like

Waiting for someone/team/community integrate Ory Hydra OAuth2 Provider

## Describe alternatives you've considered

The alternative is integrate it, and contribute with PR of the Integration with some info how to setup and a online demo of everything working, and a deployed OryHydra server if supabase members wants to test integration without deploy it

## Additional context

This same feature request will be in repos supabase-js and supabase gotrue repos

- [related issue](https://github.com/supabase/gotrue-js/issues/316)

**Comment by AaronDewes** [comment]
Having Nextcloud would be great.

[Documentation](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/oauth2.html)
[Example client in PHP](https://github.com/bahuma20/oauth2-nextcloud)

**Comment by CatalanCabbage** [comment]
# Zoho  
Hi, would be nice to have [Zoho](https://www.zoho.com/) as an OAuth provider.  
Willing to submit a PR too, please let me know if I can :)

[Documentation](https://www.zoho.com/accounts/protocol/oauth.html)

**Comment by J0** [comment]
## Feature request

Support Game Center OAuth

## Additional context

https://github.com/supabase-community/supabase-flutter/issues/156
https://developers.google.com/games/services

**Comment by nerixim** [comment]
# LINE

Hi, would be nice to have [LINE](https://developers.line.biz/en/reference/line-login/) as an OAuth provider.
Gonna try submitting a PR 💪

**Comment by yuta-51** [comment]
## Battle.net
I'd like to have [Battle.net](https://www.battle.net) as an OAuth provider.
Thank you!


[Documentation](https://develop.battle.net/documentation/guides/using-oauth)

**Comment by point-source** [comment]
# Amazon.com

It would be great to have sign in with Amazon as I have a client waiting on support before purchasing. Happy to sponsor an effort to get this done.

[Here is the documentation.](https://developer.amazon.com/apps-and-games/login-with-amazon)

**Comment by pythonicode** [comment]
# Garmin Connect

Optionally, another important connection for sport-related apps. I don't know it it supports OAuth2 but seems to support OAuth1 but it looks more complicated to get setup than Strava.

Docs: https://developer.garmin.com/gc-developer-program/program-faq/

**Comment by J0** [comment]
## Provider request: Mastodon

Is your feature request related to a problem? Please describe.

## Mastodon is growing in popularity and could be very useful to add as an auth provider.

## Additional context

Due to the federated nature of mastodon, I think login can be setup per-server instance.

I'm not exactly sure how to reflect that.

**Comment by AinurY300** [comment]
Please add the possibility of OAuth via Yandex ID and VK ID

Yandex ID: [click](https://yandex.ru/id/business?utm_source=yandex&utm_medium=yandex-id&utm_campaign=link-owners-sites)
VK ID: [click](https://platform.vk.com/vkid)

These services are the largest authorization providers in the CIS countries

**Comment by AinurY300** [comment]
----------
### **Yandex ID and VK ID**
----------
Please add the possibility of OAuth via Yandex ID and VK ID

- Yandex ID: [click](https://yandex.ru/id/business?utm_source=yandex&utm_medium=yandex-id&utm_campaign=link-owners-sites)
- VK ID: [click](https://platform.vk.com/vkid)

These services are the largest authorization providers in the CIS countries

**Comment by keepant** [comment]
## Provider Request: Huawei
It would be nice if users could authenticate with Huawei Auth
Thank you.

## Additional context
Documentation: [Huawei Auth](https://developer.huawei.com/consumer/en/agconnect/auth-service/)

**Comment by rohanliston** [comment]
# Add generic OAuth2/OIDC provider

A generic OAuth2/OIDC provider similar to what [Auth0 offers](https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/oauth2) would cover most (if not all) of the provider requests in this thread in one hit.

This would enable developers to simply provide, at a minimum:
* Authorisation Server URL (which provides a `.well-known/openid-configuration` endpoint to discover token URLs etc).
* Client ID
* Client Secret
* Callback URL
* (maybe) a mapping for claims?

I'm surprised this hasn't already been suggested. Is there some complexity that I'm missing?

**Comment by kangmingtay** [comment]
Hey @rohanliston, great question! not all social providers comply to the OIDC spec and might not have an authorization server URL. As for the mapping for claims, not all the social providers return the user data the same way (take linkedin vs workos for e.g.). Also, gotrue currently follows an automatic linking model to link 2 identities with the same email to the same user. This means that having an insecure oauth provider added could compromise logins for a user. This is also why we currently enforce the email returned from the oauth provider to be verified.

**Comment by kangmingtay** [comment]
## Add Yahoo as OAuth provider



Support Yahoo oauth.

## Describe the solution you'd like

See feature request <https://github.com/supabase/gotrue/issues/1191>

**Comment by rohanliston** [comment]
@kangmingtay Thanks for your response! To address your points:

> not all social providers comply to the OIDC spec and might not have an authorization server URL.

That's fine, there are still countless providers that are OIDC-compliant. The ones that aren't compliant would require a dedicated provider as they do now. I don't think this is a reason not to implement a generic provider.

> As for the mapping for claims, not all the social providers return the user data the same way (take linkedin vs workos for e.g.). 

Yep, we'd definitely need a mapper of some sort. A simple approach would be to have the user define a JSON object to define the mapping. Dot notation could be used to map nested fields.

> Also, gotrue currently follows an automatic linking model to link 2 identities with the same email to the same user. This means that having an insecure oauth provider added could compromise logins for a user. This is also why we currently enforce the email returned from the oauth provider to be verified.

Does gotrue require the user to authenticate to both accounts before linking? Relying on email verification alone isn't totally secure, because the account can still be compromised later on. Email verification only proves that the user had control of the account when it was created.

Auth0 highlights this in their [docs](https://auth0.com/docs/manage-users/user-accounts/user-account-linking#precautions) and has a secure approach to account linking in their [Account Link Extension](https://auth0.com/docs/customize/extensions/account-link-extension#how-does-the-extension-work-):

```
The extension does not automatically link users with the same email, even if emails are verified,
because verified emails are not enough evidence to prove that the user can currently authenticate
to both accounts.

...

This process [of authenticating to both accounts at link time] ensures that the user has the credentials
to authenticate to both accounts, which allows the accounts to be linked safely and correctly.
```

Regardless, I think whether or not an external provider is 'secure' should be a concern of the application developer setting up the integration. Gotrue should merely provide the means to support the integration itself in a secure way within its own sphere of control.

**Comment by paustint** [comment]
# Support Salesforce as an OAuth Provider
Currently, Supabase does not support Salesforce as an external OAuth provider.

# Describe the solution you'd like
Suport Salesforce as an OAuth provider https://login.salesforce.com/.well-known/openid-configuration

# Describe alternatives you've considered
N/A

# Additional context
Well-known config: https://login.salesforce.com/.well-known/openid-configuration
Docs: https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_web_server_flow.htm&type=5

**NOTE**: Salesforce allows users to configure custom subdomains for their environments, but support for this would not be required as the user can go to the standard login page (login.salesforce.com and login their, unless explicitly disallowed, in which case they can choose "Use Custom Domain".

In order to test this flow:
1. Get a free developer edition org - https://developer.salesforce.com/signup
2. In setup menu, go to "App Manager" and choose "New Connected App"
3. Enter required fields and select "Enable OAuth Settings"
4. Specify callback URL and choose some default scopes (scopes can be overridden at runtime) and save
5. Click Manage Consumer Details to see the client id and client secret

![image](https://github.com/supabase/gotrue/assets/5461649/1eebc812-809d-47ba-8964-0862048b44a2)

![image](https://github.com/supabase/gotrue/assets/5461649/5d71da82-f560-4d5b-b965-da8926c91917)

**Comment by carlobeltrame** [comment]
> # Add generic OAuth2/OIDC provider

https://github.com/supabase/gotrue/issues/451#issuecomment-1641660848

I'd like to work on a generic OAuth2 provider. Since the comments of @rohanliston in August, @kangmingtay has updated the CONTRIBUTING.md text to suggest that such a generic provider is officially regarded as a possible way forward.

By default, I would go for a generic OAuth2 client, similar to the one described by the [auth0 docs](https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/oauth2). This is as opposed to a generic OIDC flow, which was previously present in gotrue but was then removed in #927, for reasons which aren't fully transparent to me. If the generic OAuth2 client is the wrong direction to head in, please let me know.

**Comment by MonsterDeveloper** [comment]
@J0 any updates on the Telegram provider? It has been more than a year now since its request, and as far as I can see it is one of the most requested providers in this thread.

**Comment by Mutondi** [comment]
> # Add generic OAuth2/OIDC provider
> A generic OAuth2/OIDC provider similar to what [Auth0 offers](https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/oauth2) would cover most (if not all) of the provider requests in this thread in one hit.
> 
> This would enable developers to simply provide, at a minimum:
> 
> * Authorisation Server URL (which provides a `.well-known/openid-configuration` endpoint to discover token URLs etc).
> * Client ID
> * Client Secret
> * Callback URL
> * (maybe) a mapping for claims?
> 
> I'm surprised this hasn't already been suggested. Is there some complexity that I'm missing?

Is there any progress on this?

**Comment by stripuramallu3** [comment]
## Linear
I would like [Linear](https://linear.app/) as an OAuth Provider

Documentation: https://developers.linear.app/docs/oauth/authentication

**Comment by carlobeltrame** [comment]
> Is there any progress on this?

@Mutondi I have started working on it, currently I am looking into how I can extend the database schema in order to store the additional information required for genericity, such as the field mapping.

**Comment by Whats-A-MattR** [comment]
> ## Support Steam as an external OAuth Provider
> Currently, Supabase does not support Steam as an external OAuth provider.
> 
> ## Describe the solution you'd like
> Support Steam as an external OAuth Provider. https://partner.steamgames.com/doc/features/auth
> 
> ## Describe alternatives you've considered
> N/A
> 
> ## Additional context
> This [article](https://owlspace.xyz/cybersec/steam-login/) describes how Steam's login method works.

Is there an ETA for Steam as an Auth Provider? Or even a Custom Provider option?

**Comment by jessebot** [comment]
# Support Zitadel as a provider

Please add support for [ZITADEL](https://zitadel.com/). I see there is already KeyCloak support, so I could try to copy that for Zitadel, as in most instances Zitadel drops in a replacement pretty fine as both are OIDC compliant and common self hosted Identity Providers.

**Comment by carlobeltrame** [comment]
@Mutondi, @rohanliston, @kangmingtay, @bdelwood, @James3UK, @sannajammeh, @bluengreen, @jessebot, @chrisjh, @agrantdeakin, @mstade, @WildEgo, @kermado, @JoaquimLey, @naohiro-t, @BayTec, @jamiefolsom, @point-source, @Whats-A-MattR and everyone else who has mentioned or reacted to a generic OAuth provider:

I have implemented a first version of a generic OAuth provider at #1372.

If you have the means, it would help a great deal if you could test it with some real-life identity providers (even ones which are already supported by gotrue would help). I have so far tested it using an application of my own. But the more we can test this new all-purpose OAuth provider the merrier.

**Comment by rahul3399** [comment]
add miniOrange as OAuth Provider

**Comment by Wintersboy** [comment]
Would love to see Yahoo in the list of auth providers. Would make accessing the Yahoo Fantasy API so much better.

**Comment by MiryangJung** [comment]
Is there any progress on the generic OAuth provider, is possible submit  PR that adds another social login provider?

**Comment by biblebreeze** [comment]
> ## Add Yahoo as OAuth provider
> Support Yahoo oauth.
> 
> ## Describe the solution you'd like
> See feature request #1191

Any feedback on adding Yahoo oauth?
https://developer.yahoo.com/oauth2/guide/

**Comment by maurotrigo** [comment]
## Support Instagram as an external OAuth Provider

Currently, Supabase does not support Instagram as an external OAuth provider, although it does support Facebook. I understand that the flow would be very similar.

### Describe the solution you'd like
Support Instagram as an external OAuth Provider. More [here](https://developers.facebook.com/docs/instagram-basic-display-api/getting-started).

Thanks!

**Comment by mekkoo** [comment]
# Support LINE as a provider
LINE is a dominant IM app mainly in East Asia. The social login is widely used, and there are many people who want to use it with Supabase Auth.

LINE's user numbers in different countries as of 2024 are as follows:
- Japan: Approximately 95 million monthly active users
  - About 88% of Japan's adult population uses LINE
- Thailand: Approximately 51 million monthly active users
- Taiwan: Approximately 21 million monthly active users
- Indonesia: Approximately 13 million monthly active users

Other notable statistics:
- LINE's global monthly active users are estimated to be around 224 million
- The total monthly active users in the four main markets (Japan, Taiwan, Thailand, and Indonesia) is approximately 178 million
- LINE offers its services in more than 230 countries

Official document (ENG): https://developers.line.biz/en/docs/line-login/overview/
Other topic: https://github.com/orgs/supabase/discussions/4500

**Comment by ekankr2** [comment]
## Support NAVER as a provider

Please consider adding NAVER auth sign in provider.
Almost half of South Koreans use NAVER auth signin.
There is Kakao login as a provider.
I think NAVER auth provider would also be necessary too. 

Official docs : [https://developers.naver.com/products/login/api/api.md](https://developers.naver.com/products/login/api/api.md)

**Comment by purpleKarrot** [comment]
Support GitHub Apps as provider
---
Both OAuth apps and GitHub Apps use OAuth 2.0.
In general, GitHub Apps are preferred over OAuth apps.
For more information, see "[Differences between GitHub Apps and OAuth apps](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps)."

**Comment by wolfspyre** [comment]
I'd love to see Gitlab SELFHOSTED as an option for an auth provider.

**Comment by Whats-A-MattR** [comment]
> I'd love to see Gitlab SELFHOSTED as an option for an auth provider.

I'm not sure that would serve the purpose of the project. That would only be useful for internal use cases, unless you're going to write it and submit a PR I can't see time being spent on this. Sorry to be blunt.

**Comment by wolfspyre** [comment]
> > I'd love to see Gitlab SELFHOSTED as an option for an auth provider.
> 
> I'm not sure that would serve the purpose of the project. That would only be useful for internal use cases, unless you're going to write it and submit a PR I can't see time being spent on this. Sorry to be blunt.
Heya, Matt!
I hear you, and both agree and disagree with you... :)
 
There's already a Gitlab provider, I'd expect that the only difference between the two is the endpoint.

Gitlab encourages self-hosting.
Github does not. 

While many selfhosted rigs don't use legitimate ssl certs, the vast majority do.
I would **expect** that the only change needed would be to allow one to override the endpoints involved. 

If it were truly that simple, I disagree with your assertion that it would serve the purpose of the project.

if it were **NOT** that simple, (I don't know, with confidence, one way or the other, quite honestly.) I can concede the point that time/energy spent likely aren't worth the effort.
 
However, assuming it is that simple, (hey, gotta be optimistic sometimes, right?) I don't see much of a downside... altho admittedly I'm likely missing some significant facet of complication due to ignorance ;)

so... can you help me understand what I'm missing? :)

**Comment by Whats-A-MattR** [comment]
The main thing I would suggest may prevent this is that the package is focused on Social Auth. ie. Everyone can have a Facebook account, or a Twitter Account, or Google, or x ,y ,z, etc. and if the developer chooses, they can use those services to authenticate.
Not everybody (I hope?) can have an account on YOUR GitLab, which is what differentiates it from hosted GitLab. 
I can sign up for (hosted) GitLab right now, as a regular old user, and then sign in to a site made with Supabase, where GitLab is enabled as an Auth Provider using my GitLab account. 
I cannot, however, simply sign up for your hosted GitLab - at least I shouldn't be able to as per my understanding where self-hosted GitLab is a companies own thing for their people only. 

Hope that makes sense.

**Comment by wolfspyre** [comment]
> The main thing I would suggest may prevent this is that the package is focused on Social Auth. ie. Everyone can have a Facebook account, or a Twitter Account, or Google, or x ,y ,z, etc. and if the developer chooses, they can use those services to authenticate. Not everybody (I hope?) can have an account on YOUR GitLab, which is what differentiates it from hosted GitLab. I can sign up for (hosted) GitLab right now, as a regular old user, and then sign in to a site made with Supabase, where GitLab is enabled as an Auth Provider using my GitLab account. I cannot, however, simply sign up for your hosted GitLab - at least I shouldn't be able to as per my understanding where self-hosted GitLab is a companies own thing for their people only.
> 
> Hope that makes sense.


Heya Matt!

First, thanks for talking about it :)  

I appreciate the gift of your time and energy. 

Thank you.


I think this might be where our lenses diverge... 

I'll elucidate my perspective... my intent is not to argue or assert that one or the other is more 'right'....  Rather just trying to express the logic behind my reasoning..  (or what passes for logic between these ears ;) )

As I see it, The package is focused on facilitating external auth for a tool. 
It's common for teams to want to offload the responsibility of user auth to `$(notmyproblem)` 

`Auth` facilitates that offloading, by providing a conduit for brokering auth tokens. 

Now, a vast majority of users will absolutely lean on ubiquitously-accessible services to provide those tokens, as such, I wholly agree with you that this functionality will only increase the utility derived from the tool for a small cohort of users. 

However, that doesn't mean that there's no value in the ability to additionally support variant-instantiations of those ubiquitously-available services' facades for less-ubiquitously-available auth-brokers.   
**So long as it behaves the same**  
Admittedly, that's a significant caveat... and may cause more problems than it solves, as the support burden around snowflakes is .....   
well... I don't need to tell you... :) you clearly already know :) 

Nevertheless, Is the use-case for `Auth` ONLY for environments which are exposed to the public Internet?

I'd think that's kinda short-sighted, in that `Auth` shouldn't **REALLY** care who it's getting auth from, as 
Its role is simply to be the abstraction layer between `${APP-DESIRING-AUTH-VALIDATION}` and `${API-PERFORMING-AUTH-VALIDATION}`; right?

The audience diversity of `${API-PERFORMING-AUTH-VALIDATION}` isn't really relevant, is it? 

Allowing customers to use their on-prem services to legitimize their users seems like a thing that devs would derive value from, rather than being burdened by.... 

I was simply figuring that an upside would be a reduction in external network traffic, latencies around auth, as well as greater flexibility offered to app devs to subsequently offer to their customers..

¯\\\_(ツ)_/¯ 

So.. that's the lens I was using when proposing this... 
 if the expectation is that `Auth` only provides value to tools that wish to use cloud-connected services, I can totally see why the idea has no merit.

**Comment by Whats-A-MattR** [comment]
> However, that doesn't mean that there's no value in the ability to additionally support variant-instantiations of those ubiquitously-available services' facades for less-ubiquitously-available auth-brokers. **So long as it behaves the same** Admittedly, that's a significant caveat... and may cause more problems than it solves, as the support burden around snowflakes is ..... well... I don't need to tell you... :) you clearly already know :)

Yeah, quite burdensome - especially when each installation may differ and having to support different environments complicates things greatly. 

> Nevertheless, Is the use-case for `Auth` ONLY for environments which are exposed to the public Internet?

The issue I foresee isn't related to cloud or no, it's purely around number of uses that will make use of it. 

> I'd think that's kinda short-sighted, in that `Auth` shouldn't **REALLY** care who it's getting auth from, as Its role is simply to be the abstraction layer between `${APP-DESIRING-AUTH-VALIDATION}` and `${API-PERFORMING-AUTH-VALIDATION}`; right?

Completely correct, agreed. 

> The audience diversity of `${API-PERFORMING-AUTH-VALIDATION}` isn't really relevant, is it?

Only in terms of priority, very few users would equate to low priority. 


> So.. that's the lens I was using when proposing this... if the expectation is that `Auth` only provides value to tools that wish to use cloud-connected services, I can totally see why the idea has no merit.

I think there is a slight miscommunication in my message. I am not saying that such an integration would be useless, more than because of the specificity of it's nature and the small cohort that may end up using this, allocating dev time may be a difficult ask. 

I completely agree there is value in it, but it's substantially less value than a larger, less bespoke integration. 

I think dev time would be better spent on 'generic' and extensible connectors rather than bespoke solutions for on prem. 
Ie. Generic SAML (already exists, could be better), Generic OIDC, Generic OAuth - therefore enabling users to integrate whatever suits their needs.

**Comment by wolfspyre** [comment]
concur on all. Glad to have interacted ;) 

Thanks for your perspective and input... 

Totes agree on priority, too. I was mostly just adding the request as if it's not made, there's no way for y'all to gauge interest

Thanks for reducing toil for devs. Its appreciated. 
You're apprecaited.

❤️🐺w

**Comment by AlexAntonik** [comment]
### **Support Google Play Games as provider**
Almost every game on Android support Play Games Auth. Seamless(no login screen) authorization is musthave for games.

From my point of view flow is very similar for standart auth via google. 
[Probably useful info for integration](https://developer.android.com/games/pgs/android/server-access#send_the_server_auth_code)

Some privious reuests:
[Game Center and Google Play Services integration #156](https://github.com/supabase/supabase-flutter/issues/156)

#601

**Comment by goszczynskip** [comment]
### Support AT Protocol (incl. Bluesky) as a provider
This will allow to build applications on top of AT Protocol easily. 

https://docs.bsky.app/blog/oauth-atproto
https://docs.bsky.app/docs/advanced-guides/oauth-client

**Comment by mibressler** [comment]
> ### Support AT Protocol (incl. Bluesky) as a provider
> This will allow to build applications on top of AT Protocol easily.
> 
> https://docs.bsky.app/blog/oauth-atproto https://docs.bsky.app/docs/advanced-guides/oauth-client

AGREE!

**Comment by MrBE4R** [comment]
## Support Authentik as an external OAuth Provider
 
## Describe the solution you'd like
Support Authentik as an external OAuth Provider.

## Describe alternatives you've considered
N/A

## Additional context
This [article](https://docs.goauthentik.io/docs/add-secure-apps/providers/oauth2/) describes Authentik OAuth2 providers.

**Comment by typed-sigterm** [comment]
### QQ as OAuth provider

QQ is the second most widely used social platform in China. Documents are  [here](https://wiki.connect.qq.com/oauth2-0%E7%AE%80%E4%BB%8B), thanks :heart:

**Comment by maxrpark** [comment]
## Support Naver as Provider

According to recent information, Naver has approximately 40 million active users in South Korea, making it the dominant search engine in the country.

Having Kakao as auth provider is great but naver is more than necesary.

**Comment by pharesdiego** [comment]
## Support Tiktok as an external OAuth Provider

Currently, Supabase does not support Titktok as an external OAuth provider.

## Describe the solution you'd like
Support Tiktok as an external OAuth Provider. https://developers.tiktok.com/doc/login-kit-desktop

## Describe alternatives you've considered
N/A

## Additional context
This [article](https://developers.tiktok.com/doc/login-kit-overview) describes how Tiktok's login method works.

**Comment by Netz00** [comment]
## Support Reddit as an external OAuth Provider

### Describe the solution you'd like

Support Reddit as an external OAuth Provider.

### Describe alternatives you've considered

N/A

### Additional context

This [article](https://github.com/reddit-archive/reddit/wiki/OAuth2) provides an overview of Reddit's OAuth2 providers. While the resource appears outdated, it remains functional.

**Comment by SJunwikul** [comment]
## Support ARES as an external OAuth Provider
 
### Describe the solution you'd like
Support ARES as an external OAuth Provider. This is necessary for partners to integrate ARES as a sign in option and the access token is required to call the IdP's APIs.

### Describe alternatives you've considered
Using a third-party auth like Auth0 but that messes with the existing supabase auth and there's barely any documentation from Supabase about integrating Auth0. It doesn't work.

### Additional context
https://pantheon-technologies.gitbook.io/ares-for-partners
https://joinares.com/partners

**Comment by thealishakeri** [comment]
**Support Tiktok as an external OAuth Provider**
Currently, Supabase does not support Titktok as an external OAuth provider.

**Describe the solution you'd like**
Support Tiktok as an external OAuth Provider. https://developers.tiktok.com/doc/login-kit-desktop

**Describe alternatives you've considered**
N/A

**Additional context**
This [article](https://developers.tiktok.com/doc/login-kit-overview) describes how Tiktok's login method works.

https://github.com/supabase/auth/pull/269 did implement Tiktok as a provider but was never merged because it was supposed to be opened as an individual PR for better readability and testing

**Comment by jonkristian** [comment]
#### Vipps OAuth Provider

Add support for Vipps, the no. 1 mobile payment provider in Norway - https://developer.vippsmobilepay.com/docs/APIs/login-api/

#### Additional context
https://github.com/orgs/supabase/discussions/14093

**Comment by kayacancode** [comment]
How about for Instagram?

**Comment by golbin** [comment]
Is there any update or rough timeline for this?

This issue has been open for a while, and having even a rough idea of the support timeline would be really helpful for Supabase users to plan their own product roadmaps accordingly.

**Comment by dhamanutd** [comment]
@thealishakeri I create PR again for TikTok, but no reviews from the team.
#1807

**Comment by kyleorin** [comment]
how about Snapchat as an authentication method ?

**Comment by dhunt-xsolla** [comment]
> Steam OAuth2

+1; and related: https://github.com/supabase/auth/issues/451

**Comment by marcospgp** [comment]
this thread has been around for ~3 years, has any platform been added to the auth product as a result of it yet?

also am I crazy or is this the worst format for having a community vote on requests? lol there's a lot of comments to sift through

**Comment by cstockton** [comment]
Hi everyone, I apologize for the lack of updates here. We have added additional providers but there are so many that users want. We decided that the solution would be generic support for any standard OIDC / OAuth provider. We currently have this on our roadmap, tentatively planned for around January 2026. This is subject to change and discussions like this help us prioritize so thanks everyone for chiming in.

**Comment by dylanh724** [comment]
🤔 Not sure how compliant Steam is with this. And the simplicity of the prebuilts is one of the reasons we chose Supa and not raw Aws.

Generalized should be a feature of its own option. Not a replacement for the "main" ones. Steam should be a main auther.

**Comment by Abhiigg** [comment]
Zitadel

**Comment by ddtch** [comment]
please can we add [Telegram](https://core.telegram.org/api/auth) support?

**Comment by tessamerrill** [comment]
> Hi everyone, I apologize for the lack of updates here. We have added additional providers but there are so many that users want. We decided that the solution would be generic support for any standard OIDC / OAuth provider. We currently have this on our roadmap, tentatively planned for around January 2026. This is subject to change and discussions like this help us prioritize so thanks everyone for chiming in.

@cstockton Is there any update on this?

**Comment by cstockton** [comment]
Yes, this is now live with our custom OIDC / OAuth feature! For more info see the docs at https://supabase.com/docs/guides/auth/custom-oauth-providers

**Comment by cstockton** [comment]
Forgot to tag you above @tessamerrill
